applications/nrf5340_audio/src/bluetooth/bt_management/CMakeLists.txt

```diff
@@ -8,11 +8,22 @@ target_sources(app PRIVATE
 	       ${CMAKE_CURRENT_SOURCE_DIR}/bt_mgmt.c
 )
 
-if (CONFIG_BT_CENTRAL OR CONFIG_BT_OBSERVER)
+if (CONFIG_BT_CENTRAL)
 	target_sources(app PRIVATE
-	       ${CMAKE_CURRENT_SOURCE_DIR}/bt_mgmt_scan.c)
+		${CMAKE_CURRENT_SOURCE_DIR}/bt_mgmt_scan_for_conn.c)
 endif()
 
+if (CONFIG_BT_BAP_SCAN_DELEGATOR)
+target_sources(app PRIVATE
+		${CMAKE_CURRENT_SOURCE_DIR}/bt_mgmt_scan_for_broadcast.c)
+endif()
+
+if (CONFIG_BT_OBSERVER)
+target_sources(app PRIVATE
+	        ${CMAKE_CURRENT_SOURCE_DIR}/bt_mgmt_scan.c)
```

Comment:
The application only has to include bt_mgmt.h. Where the functions live underneath should not matter since this is handled by CMakeLists.txt. By having the function named bt_mgmt_scan_start() it might as well have been in bt_mgmt.c. The reason it is not is because it should only be compiled if the BT_OBSERVER role is set.